File: CivitasJ/src/civitas/CasillaJuez.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

import java.util.ArrayList;

/**
 *
 * @author ELVIRA
 */
public class CasillaJuez extends Casilla{
    
    
    /** 
     * @brief constructor de casilla juez
     * @param numcarcel numero de la casilla carcel
     * @param nombre el nombre de la casilla
     */ 
    CasillaJuez (int numCarcel, String nombre){
        super(nombre);
        setCarcel(numCarcel);
        //tipo = TipoCasilla.JUEZ;
    }
    
    /**
     * @brief Se encarcela al jugador actual
     * @param actual
     * @param todos
     */
    @Override
    void recibeJugador(int actual, ArrayList<Jugador>todos){
        if(jugadorCorrecto(actual, todos)){
            System.out.println("El Juez te manda a la carcel");
            informe(actual,todos);
            todos.get(actual).encarcelar(Casilla.getCarcel());  
        }
    }
            
    
    /**
     * @brief represente con detalle la información acerca de la casilla
     */
    
    @Override
    public String toString(){
        String text = super.toString();
        
        text = text +
               "\n El tipo : JUEZ";
        text = text +"\n carcel: "+ Integer.toString(getCarcel());
              
        return text;
    }   
    
}

File: CivitasR/lib/civitas/casilla_juez.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
require_relative "casilla.rb"
#require "byebug"
module Civitas
  class CasillaJuez < Casilla
    
    
#  /** 
#   * @brief constructor de casilla juez
#   * @param numcarcel numero de la casilla carcel
#   * @param nombre el nombre de la casilla
#   */ 
    def initialize (numCarcel, nombre)
      super(nombre)
      @@carcel = numCarcel
    end
    
    
#     /**
#     * @brief Se encarcela al jugador actual
#     * @param actual
#     * @param todos
#     */
    def recibeJugador( actual, todos)
        if(jugadorCorrecto(actual, todos))
            informe(actual,todos)
            todos.at(actual).encarcelar(@@carcel)
        end
    end 
    
    #    /**
#     * @brief represente con detalle la información acerca de la casilla
#     */
    def toString()
      text = " \n nombre: " + @nombre.to_s
      text =text+    " \n El tipo :   Juez" +
                    " \n carcel: "+ @@carcel.to_s
        return text;
    end   
    
    
    
  end
end